src/stdune/id.mli

```diff
@@ -0,0 +1,44 @@
+open !Set
+
+type t
+
+(** Convert an ID to an integer. *)
+val to_int : t -> int
+
+(** The internal IdGen module which generates an
+    id using an instance of an ID generator for
+    the next ID. *)
+module IdGen : sig
+  type idgen 
+
+  val peek : idgen -> t
+  val create : unit -> idgen
+  val to_int : t -> int
+  val gen : idgen -> t
+end
+
+module Set : Set_intf.S with type elt = t
+
+module type IdMod = sig
```

Comment:
The usual convention is to call this `S`.

Reply:
updated